src/corehost/cli/fxr/hostfxr.cpp

```diff
@@ -396,55 +397,454 @@ SHARED_API hostfxr_error_writer_fn hostfxr_set_error_writer(hostfxr_error_writer
+// property value for the active host context.
+//
+SHARED_API int32_t __cdecl hostfxr_get_runtime_property_value(
+    const hostfxr_handle host_context_handle,
+    const pal::char_t *name,
+    /*out*/ const pal::char_t **value)
+{
+    if (name == nullptr || value == nullptr)
+        return StatusCode::InvalidArgFailure;
+
+    trace::setup();
+    trace::info(_X("--- Invoked hostfxr_get_runtime_property_value [commit hash: %s]"), _STRINGIFY(REPO_COMMIT_HASH));
+
+    const host_context_t *context;
+    if (host_context_handle == nullptr)
+    {
+        const host_context_t *context_maybe = fx_muxer_t::get_active_host_context();
+        if (context_maybe == nullptr || context_maybe->type != host_context_type::active)
+        {
+            trace::error(_X("Hosting components context has not been initialized. Cannot get runtime properties."));
+            return StatusCode::HostInvalidState;
+        }
+
+        context = context_maybe;
+    }
+    else
+    {
+        context = static_cast<const host_context_t*>(host_context_handle);
+    }
+
+    if (context->type == host_context_type::invalid)
+        return StatusCode::InvalidArgFailure;
+
+    if (context->type == host_context_type::secondary)
+    {
+        const std::unordered_map<pal::string_t, pal::string_t> &properties = context->config_properties;
+        auto iter = properties.find(name);
+        if (iter == properties.cend())
+            return StatusCode::HostPropertyNotFound;
+
+        *value = (*iter).second.c_str();
+        return S_OK;
+    }
+
+    assert(context->type == host_context_type::initialized || context->type == host_context_type::active);
+    const corehost_context_contract contract = context->context_contract;
+    return contract.get_property_value(contract.instance, name, value);
+}
+
+//
+// Sets the value of a runtime property for an initialized host context
+//
+// Parameters:
+//     host_context_handle
+//       Handle to the initialized host context
+//     name
+//       Runtime property name
+//     value
+//       Value to set
+//
+// Return value:
+//     The error code result.
+//
+// Setting properties is only supported for the first host context, before the runtime has been loaded.
+//
+// If the property already exists in the host context, it will be overwritten. If value is nullptr, the
+// property will be removed.
+//
+SHARED_API int32_t __cdecl hostfxr_set_runtime_property_value(
+    const hostfxr_handle host_context_handle,
+    const pal::char_t *name,
+    const pal::char_t *value)
+{
+    if (host_context_handle == nullptr || name == nullptr)
+        return StatusCode::InvalidArgFailure;
+
+    trace::setup();
+    trace::info(_X("--- Invoked hostfxr_set_runtime_property_value [commit hash: %s]"), _STRINGIFY(REPO_COMMIT_HASH));
+
+    host_context_t *context = static_cast<host_context_t*>(host_context_handle);
+    if (context->type != host_context_type::initialized)
```

Comment:
Just like in the hostpolicy - would be nice to add a helper which converts the `hostfxr_handle` to `host_context_t`. It could include the check for invalid state.
Also like in the hostpolicy case, having a marker in the struct to have a simple check for deleted contexts would be nice as well.